### ISSUE TITLE
NPE false positive on null coalescing

### ISSUE THREAD

**Issue by SpyroTEQ** [kalessil/phpinspectionsea#1470]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.2 |
| Language level | PHP 7.2       |

Current behaviour
--
This code:
```php
function x(?\stdClass $x) {
	echo ($x->fakeField ?? false) ? $x->otherField : '';
}
x(new \stdClass());
```
Show a "Null pointer might occur here" on "$x" in "$x->otherField" (I don't really care whether "otherField" exists here, it's not the point)

Expected behaviour
--
$x cannot be null in such case, so no inspection should appear

Environment details
---
none

**Comment by kalessil** [comment]
Thank you for reporting @SpyroTEQ, fixing it!

**Comment by kalessil** [comment]
Update: did deeper tests, that's a very cool trick 👍 . And I also concerned about nullable and dynamic fields, so I have to keep the behaviour as it is, unfortunatelly.
